Vulkan-Pipeline-Assistant/Widgets/spvarraywidget.cpp

```diff
@@ -85,6 +78,35 @@ namespace vpa {
         memcpy(m_data, data, m_type->size);
     }
 
+    void SpvArrayWidget::OnDrawerInit() {
+        if (m_type->subtype->Type() == SpvTypeName::Struct) {
+            m_inputWidget->SetDrawerItemWidget(GetDrawerItemWidget());
+            m_inputWidget->OnDrawerInit();
+        }
+    }
+
+    void SpvArrayWidget::Init() {
+        m_inputWidget->Init();
+        HandleArrayElementChange();
+    }
+
+    void SpvArrayWidget::OnClick(bool expanding) {
+        if (expanding) {
+            m_container->ShowWidget(this);
+            if (m_type->subtype->Type() == SpvTypeName::Matrix || m_type->subtype->Type() == SpvTypeName::Vector) {
+                m_inputArea->ShowWidget(m_inputWidget);
+            }
+            else if (m_type->subtype->Type() == SpvTypeName::Struct) {
+                SpvWidget* defaultStructWidget = reinterpret_cast<SpvStructWidget*>(m_inputWidget)->GetTypeWidget(0);
+                if (defaultStructWidget) m_inputArea->ShowWidget(defaultStructWidget);
+            }
+        }
```

Comment:
There's a lot of indents here. Maybe consider returning early with "if(!expanding) return;"

Reply:
I agree with this, updated.